### ISSUE TITLE
[prosemirror-keymap] Shift-Alt-- triggers Alt-- (but seemingly only for every other press)

### ISSUE THREAD

**Issue by Nantris** [ProseMirror/prosemirror#1446]
I should first mention that we're using TipTap. I'm not certain how to reproduce this in ProseMirror alone because I'm not sure how to setup an editor without TipTap - **but I have very good reason to believe the issue is in ProseMirror rather than TipTap.**

I examined the TipTap code and here you can see that TipTap binds the hotkeys using ProseMirror's `keymap`:

https://github.com/ueberdosis/tiptap/blob/5a4c3baa0633105809469b20742094b01360a2be/packages/core/src/ExtensionManager.ts#L244-L254

I then set breakpoints in `prosemirror-keymap/dist/index.js` in our project and confirmed that the shortcut is being passed to `keymap` in an unaltered way (`Shift-Alt--` and `Alt--`)

But when we press `Shift-Alt--`, every other time it's pressed, it triggers the `Alt--` action.

![image](https://github.com/ProseMirror/prosemirror/assets/6835891/fffc022d-39e6-44f1-ae2a-b87d7f36f087)
![image](https://github.com/ProseMirror/prosemirror/assets/6835891/9fe6c9c8-dfd4-4aad-bd16-353300030e12)

The closest I can get to a root cause is that `direct(view.state, view.dispatch, view)` alternates between returning `true` and `false` for seemingly identical input events. I don't know why the return value differs but I hope this is enough to hint at the underlying cause.

**Comment by Nantris** [comment]
@marijnh any thoughts on this? Is there anything I can do to better clarify the problem? Unfortunately I'm not sure how to fix it despite mostly tracking it down.

**Comment by marijnh** [comment]
A plain-prosemirror reproduction would be helpful. Also information about the browser/platform you tested with. I cannot reproduce this.

**Comment by Nantris** [comment]
Thanks for your reply @marijnh and for trying to reproduce. Apologies for the lack of info - I produced this issue in Chrome and Electron.

Is there any ProseMirror CodeSandbox or the like that I could consult to start with? I regret to say that I am not sure how to produce a pure ProseMirror repro from scratch but if there's a boilerplate I'm sure I could figure it out eventually.

**Comment by marijnh** [comment]
Following the [basic example](https://prosemirror.net/examples/basic/) should make it rather straightforward to set up an editor.

**Comment by Nantris** [comment]
Thanks for the tip @marijnh.

I adapted an existing sandbox I found so it could be more minimal, but the relevant code is easy to find and isolate https://codesandbox.io/p/sandbox/gallant-albattani-pcfh6s

If you open up the console you'll see when you press `Shift-Alt-Minus` (`Shift-Alt--`) it prints for the `Alt-Minus` command as well. I couldn't replicate the particular finding that it only happened on every other press, but hopefully this repro helps to expose the overarching issue, and maybe the other issue would be resolved for us by coincidence.

![image](https://github.com/ProseMirror/prosemirror/assets/6835891/14219167-48f9-4c06-9854-54ff789f1607)
